narwhals/_pandas_like/dataframe.py

```diff
@@ -335,6 +318,25 @@ def generate_unique_token(
                     ),
                 )
 
+        if how == "anti":
+            indicator_token = generate_unique_token(
+                n_bytes=8, columns=[*self.columns, *other.columns]
+            )  # pragma: no cover
+
+            return self._from_dataframe(
+                self._dataframe.merge(
+                    other._dataframe.loc[:, right_on].rename(
+                        columns=dict(zip(right_on, left_on))  # type: ignore[arg-type]
+                    ),
+                    how="outer",
+                    indicator=indicator_token,
+                    left_on=left_on,
+                    right_on=left_on,
+                )
+                .loc[lambda t: t[indicator_token] == "left_only"]
+                .drop(columns=[indicator_token])
+            )
```

Comment:
I'm not 100% sure this is the same - could you include a test in which you anti-join on a single column which contains duplicates please?

Reply:
Just to give you a snippet of intermediate values:

```py
other = df_raw.loc[lambda t: t["b"]>5, "b"]
df_raw.merge(other, how="outer", indicator=True, left_on=["b"], right_on=["b"] )
```
```terminal
   a  b    z     _merge
0  1  4  7.0  left_only
1  3  4  8.0  left_only
2  2  6  9.0       both
```
```py
other = df_raw.loc[lambda t: t["b"]<5, "b"]
df_raw.merge(other, how="outer", indicator=True, left_on=["b"], right_on=["b"])
```

```terminal
   a  b    z     _merge
0  1  4  7.0       both
1  1  4  7.0       both
2  3  4  8.0       both
3  3  4  8.0       both
4  2  6  9.0  left_only
```

The latter could blow up memory?!
That said, I personally use anti join only if I need to filter on multiple columns i.e. tuple of values, otherwise `~ Expr.is_in(...)` is all you need.

Reply:
ok thanks - maybe this is correct then (I think it's for semi-join that duplicates may introduce issues)

but yes, I think there is a risk of memory blow-up, though arguably that's always a risk with pandas...maybe we just need a note cautioning users about using this?

Reply:
Should we drop duplicates ourselves in `other`?

```py
(other._dataframe
 .loc[:, right_on]
.rename(columns=dict(zip(right_on, left_on)))
.drop_duplicates(ignore_index=True)
)
```

Reply:
yeah that's probably be better 👍